rust/otap-dataflow/crates/pdata/src/otap/filter/traces.rs

```diff
@@ -0,0 +1,1094 @@
+    }
+}
+
+impl TraceMatchProperties {
+    /// create a new TraceMatchProperties
+    #[must_use]
+    pub fn new(
+        match_type: MatchType,
+        resource_attributes: Vec<KeyValue>,
+        span_attributes: Vec<KeyValue>,
+        span_names: Vec<String>,
+        event_names: Vec<String>,
+        event_attributes: Vec<KeyValue>,
+        link_attributes: Vec<KeyValue>,
+    ) -> Self {
+        Self {
+            match_type,
+            resource_attributes,
+            span_attributes,
+            span_names,
+            event_names,
+            event_attributes,
+            link_attributes,
+        }
+    }
+
+    /// create filter takes a traces_payload and returns the filters for each of the record batches, also takes a invert flag to determine if the filters will be inverted
+    pub fn create_filters(
+        &self,
+        traces_payload: &OtapArrowRecords,
+        invert: bool,
+    ) -> Result<(
+        BooleanArray,
+        BooleanArray,
+        BooleanArray,
+        BooleanArray,
+        BooleanArray,
+        BooleanArray,
+    )> {
+        let (
+            mut resource_attr_filter,
+            mut span_filter,
+            mut span_attr_filter,
+            mut span_event_filter,
+            mut span_event_attr_filter,
+            mut span_link_attr_filter,
+        ) = (
+            self.get_resource_attr_filter(traces_payload)?,
+            self.get_span_filter(traces_payload)?,
+            self.get_span_attr_filter(traces_payload)?,
+            self.get_span_event_filter(traces_payload)?,
+            self.get_span_event_attr_filter(traces_payload)?,
+            self.get_span_link_attr_filter(traces_payload)?,
+        );
+
+        // invert flag depending on whether we are excluding or including
+        if invert {
+            resource_attr_filter =
+                arrow::compute::not(&resource_attr_filter).expect("not doesn't fail");
+
+            span_filter = arrow::compute::not(&span_filter).expect("not doesn't fail");
+
+            span_attr_filter = arrow::compute::not(&span_attr_filter).expect("not doesn't fail");
+
+            span_event_filter = arrow::compute::not(&span_event_filter).expect("not doesn't fail");
+
+            span_event_attr_filter =
+                arrow::compute::not(&span_event_attr_filter).expect("not doesn't fail");
+
+            span_link_attr_filter =
+                arrow::compute::not(&span_link_attr_filter).expect("not doesn't fail");
+        }
+
+        Ok((
+            resource_attr_filter,
+            span_filter,
+            span_attr_filter,
+            span_event_filter,
+            span_event_attr_filter,
+            span_link_attr_filter,
+        ))
+    }
+
+    /// Creates a booleanarray that will filter a resource_attribute record batch based on the
+    /// defined resource attributes we want to match
+    fn get_resource_attr_filter(&self, traces_payload: &OtapArrowRecords) -> Result<BooleanArray> {
```

Comment:
I think the code in this method is duplicated from `otap::filter::logs::LogMatchProperties::get_filter_resources`, right?

Maybe we could extract this code into a shared function to make any future maintenance easier.

Similary, could we do the same thing for `get_span_attr_filter` below and `otap::filter::logs::LogMatchProperties::get_log_attr_filter` ?